building_sim_plugins/building_sim_ign/src/door.cpp

```diff
@@ -180,91 +72,38 @@ class IGNITION_GAZEBO_VISIBLE DoorPlugin
     ignwarn << "Initializing plugin with name " << plugin_name << std::endl;
     _ros_node = std::make_shared<rclcpp::Node>(plugin_name);
 
-    RCLCPP_INFO(
-      _ros_node->get_logger(),
+    RCLCPP_INFO(_ros_node->get_logger(),
       "Loading DoorPlugin for [%s]",
       _model.Name(ecm).c_str());
+
     auto sdfClone = sdf->Clone();
```

Comment:
Bit of a shot in the dark here, but can you try to remove the Clone function and just use the original SDF pointer? I had a look at the sdformat source code and when cloning the parent is not copied, hence is set to default of nullptr.
We would need to make sure we are not editing the sdf element then, hopefully the utilities functions take care of that.

Reply:
You hit the bullseye. I now clone the sdf element ptr inside the `make()` function and work with the clone to retrieve necessary plugin parameters. The parent is then obtained from the original inside the same function. 
https://github.com/osrf/traffic_editor/blob/c518197429338d65cc5303f89819252f2da128e2/building_sim_plugins/building_sim_common/include/building_sim_common/door_common.hpp#L131